### PR TITLE
Implement image viewer

### LINKED ISSUE #161: Simple photo viewer
The current implementation converts images to WebP format. While WebP itself is great, some browsers such as Google Chrome and Safari (but not Firefox) always download it every time clicking the image, instead of opening it in the browser. It's inconvenient when we want to see enlarg images.

To prevent this a bit annoying download behavior, I'd like a simple photo viewer. I'm thinking about something like this: https://pixelfed.social/p/PixFed/89129948895383552. A similar one can be implemented in _hyperscript.

### PR DESCRIPTION
**Pull request by shuuji3**
Resolves #161

Preview: https://takahe-dev.shuuji3.xyz
(image loading might be slow due to network bandwidth/latency)

This PR adds the image viewer to show the attached image with a description.

takahe-dev shuuji3 xyz_

takahe-dev shuuji3 xyz_ (1)

### CODE REVIEW

**Comment by andrewgodwin** [comment]
We already have #362 working towards this too - @manfre can you look at this one and see what the differences are?

**Comment by manfre-lp** [comment]
This PR is similar, but handles a few things in a nicer way. Lets go with this one. I like the transparent background to show that you're scrolling the timeline (one issue I didn't resolve on my PR).

Only change I'd want to see is the caption text needs a solid background. The white text over transparent background is difficult to read.

**Comment by manfre** [comment]
This PR is similar, but handles a few things in a nicer way. Lets go with this one. I like the transparent background to show that you're scrolling the timeline (one issue I didn't resolve on my PR).

Only change I'd want to see is the caption text needs a solid background. The white text over transparent background is difficult to read.

**Comment by shuuji3** [comment]
Sorry for the duplicated work. I should have made a draft PR. I'd like to refine the logic a bit referring to your PR.

> Only change I'd want to see is the caption text needs a solid background. The white text over transparent background is difficult to read.

Thanks for reviewing. This is a good point. I'll try the black background on the bottom caption area.

**Comment by shuuji3** [comment]
I've updated the PR to ensure non-transparent background and reorganize the code structure. New commit has already been deployed to the same server at https://takahe-dev.shuuji3.xyz/.

File: templates/activities/_post.html [hunk 1]
```diff
@@ -43,8 +43,9 @@
           <div class="attachments">
             {% for attachment in post.attachments.all %}
               {% if attachment.is_image %}
-                <a href="{{ attachment.full_url.relative }}" class="image">
-                  <img src="{{ attachment.thumbnail_url.relative }}" title="{{ attachment.name }}" alt="{{ attachment.name }}" loading="lazy" />
+                <a href="{{ attachment.full_url.relative }}" class="image" target="_blank"
+                  _="on click halt the event then call imageviewer.show(me)">
+                  <img src="{{ attachment.thumbnail_url.relative }}" title="{{ attachment.name }}" alt="{{ attachment.name|default:'(no description)' }}" loading="lazy" data-original-url="{{ attachment.full_url.relative }}">
```
Comment (shuuji3) [inline templates/activities/_post.html:48]: I added the default value when the image has no description taking #336 into consideration.

File: templates/activities/_modal_image.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        -- fill information
+        set image@src to source_url
+        set image@alt to source_alt
+        set figcaption's textContent to source_alt
+
+        -- show image viewer
+        show #image-viewer with display:flex
+        transition #image-viewer's opacity from 0 to 1 over 300ms
+        js document.querySelector('#image-viewer button').focus()
```
Comment (shuuji3) [inline templates/activities/_modal_image.html:16]: Without `focus()` close button cannot be focused using a keyboard. Also, I couldn't find a working hyperscript.

Comment (shuuji3) [reply templates/activities/_modal_image.html]: OK, the hyperscript issue can be fixed by `focus() on first <#image-viewer button />`. What I need was an additional `first`.

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
Thanks for working on this!

**Comment by shuuji3** [comment]
Thank you for reviewing! 🙂

**Pull request merged**